js/src/modals/ExportAccount/exportAccount.js

```diff
@@ -0,0 +1,201 @@
+
+import { observer } from 'mobx-react';
+import React, { Component, PropTypes } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+import { newError } from '~/redux/actions';
+import { personalAccountsInfo } from '~/redux/providers/personalActions';
+import { AccountCard, Button, Portal, SelectionList } from '~/ui';
+import { Input } from '~/ui/Form';
+import { CancelIcon, CheckIcon } from '~/ui/Icons';
+import ExportStore from './exportStore';
+
+import styles from './exportAccount.css';
+
+@observer
+class ExportAccount extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    accounts: PropTypes.object.isRequired,
+    balances: PropTypes.object.isRequired,
+    newError: PropTypes.func.isRequired,
+    personalAccountsInfo: PropTypes.func.isRequired,
+    onClose: PropTypes.func.isRequired
+  };
+
+  componentWillMount () {
+    const { accounts, newError } = this.props;
+
+    this.exportStore = new ExportStore(this.context.api, accounts, newError, null);
+  }
+
+  render () {
+    const { canExport } = this.exportStore;
+
+    return (
+      <Portal
+        buttons={ [
+          <Button
+            icon={ <CancelIcon /> }
+            key='cancel'
+            label={
+              <FormattedMessage
+                id='export.accounts.button.cancel'
+                defaultMessage='Cancel'
+              />
+            }
+            onClick={ this.onClose }
+          />,
+          <Button
+            disabled={ !canExport }
+            icon={ <CheckIcon /> }
+            key='execute'
+            label={
+              <FormattedMessage
+                id='export.accounts.button.export'
+                defaultMessage='Export'
+              />
+            }
+            onClick={ this.onExport }
+          />
+        ] }
+        onClose={ this.onClose }
+        open
+        title={
+          <FormattedMessage
+            id='export.accounts.title'
+            defaultMessage='Export an Account'
+          />
+        }
+      >
+        { this.renderList() }
+      </Portal>
+    );
+  }
+
+  renderList () {
+    let { accounts } = this.props;
+
+    accounts = Object
+      .keys(accounts)
+      .map((address) => accounts[address]);
+
+    return (
+      <SelectionList
+        isChecked={ this.isSelected }
+        items={ accounts }
+        noStretch
+        renderItem={ this.renderAccount }
+      />
+    );
+  }
+
+  renderAccount = (account) => {
+    const { balances } = this.props;
+    const balance = balances[account.address];
+    const { changePassword, getPassword, onFocus } = this.exportStore;
+    const inputValue = getPassword(account);
+
+    return (
+      <AccountCard
+        account={ account }
+        balance={ balance }
+        onClick={ () => onFocus(account.address) }
```

Comment:
This is big struggle for me. I didn't want to rewrite the components since I can't easily access the account prop from exportAccount.js, but I totally know what I can do now ... I'll add spread syntax to the onClick function in AccountCard